### PR TITLE
Use HttpVersionType enum rather than string compare

### PR DESCRIPTION
**Pull request by benaadams**
Resolves https://github.com/aspnet/KestrelHttpServer/issues/1014

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/HttpVersionType.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Http
+{
+    public enum HttpVersionType
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/HttpVersionType.cs]: Can we name it `HttpVersion` or will that conflict with something else?

Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/HttpVersionType.cs]: Renamed

Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/HttpVersionType.cs]: Conflicts with the attribute though... is prefixed

**Comment by halter73** [comment]
LGTM

**Comment by cesarbs** [comment]
:ship:

**Pull request merged**